docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,328 @@
+Iris model. Meshes concern only the spatial location of data, and can
+optionally be limited to just the horizontal location (e.g. X and Y). Other
+dimensions such as time or ensemble member (and often vertical levels)
+retain their familiar structured format.
+
+The UGRID conventions themselves are designed as an addition to the existing CF
+conventions, which are at the core of Iris' philosophy.
+
+What's Different?
+=================
+
+The mesh format represents data's geographic location using an **unstructured
+mesh**. This has significant pros and cons when compared to a structured grid.
+
+.. contents::
+   :local:
+
+The Detail
+----------
+..
+    The diagram images are SVG's, so editable by any graphical software
+     (e.g. Inkscape). They were originally made in MS PowerPoint.
+
+    Uses the IBM Colour Blind Palette (see
+     http://ibm-design-language.eu-de.mybluemix.net/design/language/resources/color-library
+     )
+
+Structured Grids
+~~~~~~~~~~~~~~~~
+Assigning data to locations using a structured grid is essentially an act of
+matching coordinate arrays to each dimension of the data array. The data can
+also be represented as an area (instead of a point) by including a bounds array
+for each coordinate array.
+
+.. figure:: images/data_structured_grid.svg
+   :alt: Diagram of how data is represented on a structured grid
+
+   Data on a structured grid
+
+   :download:`full size image <images/data_structured_grid.svg>`
+
+Unstructured Meshes
+~~~~~~~~~~~~~~~~~~~
+The most basic element in a mesh is the 0-dimensional **node**: a single
+location in space. Every node in the mesh is defined by indexing the
+1-dimensional X and Y (and optionally Z) coordinate arrays (the
+``node_coordinates``) - e.g. ``(x[3], y[3])`` gives the position of the fourth
+node. Note that this means each node has its own coordinates, independent of
+every other node. Since nodes can be anywhere in the space - **unstructured** -
+the position in the array has nothing to do with spatial position.
+
+If data is assigned to node location it must be stored in a 1D array of equal
+length to the number of nodes in the mesh. ``data[3]`` is at the position:
+``(x[3], y[3])``.
+
+Data can also be assigned to higher dimensional elements - **edges**, **faces**
+or **volumes**. These elements are constructed by connecting nodes together
+using a 2-dimensional **connectivity** array. One dimension varies over each
+element, while the other dimension varies over each node that makes up that
+element; the values in the array are the node indices. E.g. we could make 2
+square faces from 6 nodes using this ``face_node_connectivity``:
+``[[0, 1, 3, 2], [2, 3, 5, 4]]``. Remember that meshes are **unstructured**, so
+there is no significance to the order of the faces in the array. Data assigned
+to a higher dimensional location must be stored in a 1D array of equal length
+to the number of elements in the mesh, e.g. ``my_data = [0.33, 4.02]`` for our
+example.
```

Comment:
I was tracking well through this page, then this paragraph completely stumped me - the concept of "node" and "element" weren't solid enough yet that I could work out what was happening.
If it's rows and columns, can we refer to rows and columns? E.g. "Each row describes an element (edge, face or volume), by giving the nodes (points) that make it up." I think that "we could make 2 square faces, each from 4 nodes" would be much easier to understand for the example too, because otherwise I have to work out where the 6 maps to, which isn't immediately obvious.

On "Remember", I'd split to a new paragraph so that I can have a break as a user by being satisfied that I got the previous concept. I'd then reword to something like

> Data assigned to the elements described by a **connectivity** array must be stored in a 1D array with one datum per element, e.g. ``my_data = [0.33, 4.02]`` for our example because we have two square elements. Remember that meshes are **unstructured**, so there is no significance to the order of the faces in the array.

Reply:
3c0e5f85bff8e008226dc9d8a88da47b9b6f4db1